Source/DafnyCore/Plugins/IExecutableBackend.cs

```diff
@@ -128,9 +128,9 @@ public abstract class IExecutableBackend {
   }
 
   /// <summary>
-  /// Perform any required cleanups after generating code with <c>Compile</c> and <c>EmitCallToMain</c>.
+  /// Perform any required processing after generating code with <c>Compile</c> and <c>EmitCallToMain</c>.
   /// </summary>
-  public virtual void OnPostCompile() { }
+  public abstract bool OnPostCompile(string dafnyProgramName, string targetFilename, TextWriter outputWriter);
```

Comment:
I think it makes more sense to keep this as an empty virtual method, so implementing classes don't have to do anything if they don't need this hook.

Reply:
Also `targetFilename` should probably be `targetDirectory`, since you're passing `targetPaths.SourceDirectory` for it.